it/controllers/DateOfBusinessTransferControllerISpec.scala

```diff
@@ -0,0 +1,37 @@
+package controllers
+
+import helpers._
+import identifiers.DateOfBusinessTransferId
+import play.api.Application
+import play.api.inject.guice.GuiceApplicationBuilder
+import play.mvc.Http.HeaderNames
+
+class DateOfBusinessTransferControllerISpec extends IntegrationSpecBase with AuthHelper with SessionStub {
+
+  override implicit lazy val app: Application = new GuiceApplicationBuilder()
+    .configure(fakeConfig())
+    .build()
+
+  val dateFieldName = "relevantDate"
+
+  class Setup extends SessionTest(app)
+
+  s" ${controllers.routes.DateOfBusinessTransferController.onSubmit()}" should {
+    s"redirect to Name of the Previous Business" in new Setup {
+      stubSuccessfulLogin()
+      stubSuccessfulRegIdGet()
+      stubAudits()
+
+      val request = buildClient("/date-of-transfer").withHttpHeaders(HeaderNames.COOKIE -> getSessionCookie(), "Csrf-Token" -> "nocheck")
+        .post(Map(
+          s"$dateFieldName.day" -> Seq("1"),
+          s"$dateFieldName.month" -> Seq("1"),
+          s"$dateFieldName.year" -> Seq("2020")
+        ))
+      val response = await(request)
+      response.status mustBe 303
+      // TODO: Change the redirected page below to the correct page
+      response.header(HeaderNames.LOCATION) mustBe Some(controllers.routes.DateOfBusinessTransferController.onPageLoad.url)
+    }
+  }
```

Comment:
We could do with some more tests for this controller - the show route isnt tested, so we should at least have a regression test to confirm it returns OK.

Other possible tests we could have are

- show() if there is a value in user answers - confirm the form pre-populates using Jsoup.parse
- show() if there is no value in user answers - confirm we get OK and the form is blank
- submit() if bad data is bound to the form - do we get a BadRequest returned?

Reply:
All the missing IT tests have been added 👍